src/extension.js

```diff
@@ -4,11 +4,35 @@ console.log("Scroll Percentage in Tab Title Extension loaded 👍");
 
 const originalTitle = document.title;
 
-window.addEventListener("scroll", () => {
-  let scrollTop = window.scrollY;
-  let docHeight = document.body.offsetHeight;
-  let winHeight = window.innerHeight;
-  let scrollPercent = scrollTop / (docHeight - winHeight);
-  let scrollPercentRounded = Math.round(scrollPercent * 100);
-  document.title = `(${scrollPercentRounded}%) ${originalTitle}`;
-}, { passive: true });
+window.addEventListener(
+  "scroll",
+  () => {
+    const windowHeight = document.body.offsetHeight;
+    const scrollTop = window.scrollY;
+    const containers = document.querySelectorAll(
+      "body,main,*[class~='content'],*[class*='Content'],*[class~='article']:first-child,article:first-child"
+    );
+
+    let mainContainer = null,
+      mainContainerSizes = null;
+
+    for (let x = containers.length - 1; x >= 0; x--) {
+      mainContainer = containers[x];
+      mainContainerSizes = mainContainer.getBoundingClientRect();
+      if (windowHeight / mainContainerSizes.height < 3) {
+        break;
+      }
+    }
+
+    let docHeight =
+      scrollTop + mainContainerSizes.top + mainContainerSizes.height;
+    let winHeight = window.innerHeight;
+    let scrollPercent = scrollTop / (docHeight - winHeight);
+    let scrollPercentRounded = Math.min(
+      Math.max(Math.round(scrollPercent * 100), 0),
+      100
+    );
```

Comment:
Can you please explain why the additional `Math.min` and `Math.max` functions?

Reply:
Never mind. I got it.
It's because we don't want to show a negative or more than 100 percent if the content is not spending over the full page.